pkg/asset-manager-utils/test/SinglePoolAaveATokenAssetManager.test.ts

```diff
@@ -0,0 +1,330 @@
+      daiAToken.address,
+      aaveRewardsController.address,
+      stkAave.address,
+    ],
+  });
+  const assetManagers = [assetManager.address, other.address];
+
+  // Deploy Pool
+  const args = [
+    vault.address,
+    'Test Pool',
+    'TEST',
+    tokens.addresses,
+    [fp(0.5), fp(0.5)],
+    assetManagers,
+    fp(0.0001),
+    0,
+    0,
+    admin.address,
+  ];
+
+  const pool = await deploy('v2-pool-weighted/WeightedPool', {
+    args,
+  });
+
+  const poolId = await pool.getPoolId();
+
+  // Deploy staking contract for pool
+  const stakingContract = await deploy('v2-distributors/MultiRewards', {
+    args: [vault.address, pool.address],
+  });
+
+  await assetManager.initialise(poolId, stakingContract.address);
+
+  const rewardsDuration = 1; // Have a neglibile duration so that rewards are distributed instantaneously
+  await stakingContract.addReward(stkAave.address, assetManager.address, rewardsDuration);
+
+  await tokens.mint({ to: lp, amount: tokenInitialBalance });
+  await tokens.approve({ to: vault.address, from: [lp] });
+
+  const assets = tokens.addresses;
+
+  await vault.connect(lp).joinPool(poolId, lp.address, lp.address, {
+    assets: tokens.addresses,
+    maxAmountsIn: Array(assets.length).fill(MAX_UINT256),
+    fromInternalBalance: false,
+    userData: encodeJoinWeightedPool({
+      kind: 'Init',
+      amountsIn: Array(assets.length).fill(tokenInitialBalance),
+    }),
+  });
+
+  return {
+    data: {
+      poolId,
+    },
+    contracts: {
+      assetManager,
+      lendingPool,
+      tokens,
+      pool,
+      stakingContract,
+      stkAave,
+      vault,
+    },
+  };
+};
+
+describe('Single Pool Aave AToken asset manager', function () {
+  let tokens: TokenList,
+    pool: Contract,
+    vault: Contract,
+    lendingPool: Contract,
+    assetManager: Contract,
+    stakingContract: Contract,
+    stkAave: Contract;
+  let lp: SignerWithAddress, other: SignerWithAddress;
+  let poolId: string;
+
+  before('deploy base contracts', async () => {
+    [, , lp, other] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('set up asset manager', async () => {
+    const { data, contracts } = await setup();
+    poolId = data.poolId;
+
+    pool = contracts.pool;
+    assetManager = contracts.assetManager;
+    lendingPool = contracts.lendingPool;
+    stakingContract = contracts.stakingContract;
+    stkAave = contracts.stkAave;
+    tokens = contracts.tokens;
+    vault = contracts.vault;
+  });
+
+  describe('deployment', () => {
+    it('different managers can be set for different tokens', async () => {
+      expect((await vault.getPoolTokenInfo(poolId, tokens.DAI.address)).assetManager).to.equal(assetManager.address);
+      expect((await vault.getPoolTokenInfo(poolId, tokens.MKR.address)).assetManager).to.equal(other.address);
+    });
+  });
+
+  describe('setPoolConfig', () => {
+    it('allows a pool controller to set the desired target investable %', async () => {
```

Comment:
```suggestion
    it.skip('allows a pool controller to set the desired target investable %', async () => {
```

`.skip` makes it really obvious that some tests are not really running. Similarly, I'd create another (empty for now) `it` that does 'reverts if caller is not authorized'.

Reply:
well, this test does actually setPoolConfig.  I added an `xit` for the reverts if caller not authorized test.

Reply:
Huh you're right, just skipping the other one ought to be enough.